kingpin/actors/rightscale/alerts.py

```diff
@@ -678,6 +678,8 @@ def _precache(self):
         all_resource_specs = yield self._client.find_by_name_and_keys(
             self._client._client.alert_specs, exact=True,
             subject_href=self.option('href'))
+        if not isinstance(all_resource_specs, list):
```

Comment:
We keep doing this. We've got to remove the `len` check inside of `find_by_name_and_keys` - it's just a bad practice to return different data for same query